framework/transport/ieee1905_transport/ieee1905_transport_packet_processing.cpp

```diff
@@ -225,6 +225,11 @@ bool Ieee1905Transport::de_fragment_packet(Packet &packet)
 
     // concat the fragment body (excluding the IEEE1905 header)
     size_t fragmentTlvsLength = packet.payload.iov_len - sizeof(Ieee1905CmduHeader);
+    // Only count end of message TLV for the last fragment
+    if (ch->GetLastFragmentIndicator() == 0) {
+        fragmentTlvsLength -= sizeof(Tlv);
```

Comment:
This is potentially problematic. Since short frames may be padded, `packet.payload.iov_len` is not an exact indicator of the CMDU length. Instead, the TLVs should be parsed to vind the EOM TLV.

That said, it's *very* unlikely that any fragment except for the last one will be a short frame, so (at least for now) it's OK as it is.

Maybe we need to create an issue for this though.

Reply:
You are right, it is a real issue and can easily happen as we have VS TLVs which contain only an ACTION_OP (uint). So if we build a message which has such a TLV followed by a really big one, this will reproduce. However, it is a corner case that we don't encounter at the moment, so we can fix it at our own pace. It is actually a very good first issue, since it ramps someone up on the whole architecture.

Opened a task for this - #86